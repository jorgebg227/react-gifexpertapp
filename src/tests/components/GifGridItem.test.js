import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
    const title = 'Un t√≠tulo';
    const url = 'http://localhost:3000/logo192.png';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('debe de mostrar <GifGridItem /> correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el titulo', ()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('debe de tener la imagen con el url y el alt igual a los props ', () => {
        const img = wrapper.find('img');
        //console.log(img.props())
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener la clase animate__backInLeft', ()=>{
        const div = wrapper.find('div');
        const animate='animate__backInLeft';
        const className = div.prop('className');
        expect(className.includes(animate)).toBe(true);
    });
    
});
