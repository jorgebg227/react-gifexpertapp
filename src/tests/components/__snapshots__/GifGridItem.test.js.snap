// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img
    alt="Un título"
    src="http://localhost:3000/logo192.png"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
